graphql_compiler/macros/__init__.py

```diff
@@ -212,18 +207,7 @@ def get_schema_for_macro_definition(schema):
     Raises:
         AssertionError, if the schema contains directive names that are non-default.
     """
-    macro_definition_schema = copy(schema)
-    macro_definition_schema_directives = schema.get_directives()
-    check_for_nondefault_directive_names(macro_definition_schema_directives)
-    macro_definition_schema_directives += DIRECTIVES_REQUIRED_IN_MACRO_EDGE_DEFINITION
-    # Remove disallowed directives from directives list
-    macro_definition_schema_directives = list(set(macro_definition_schema_directives) &
-                                              set(DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION))
-
-    # pylint: disable=protected-access
-    macro_definition_schema._directives = macro_definition_schema_directives
-    # pylint: enable=protected-access
-    return macro_definition_schema
+    return get_schema_for_macro_edge_definitions(schema)
```

Comment:
I do not remove `get_schema_for_macro_edge_definitions` and  only keep `get_schema_for_macro_definitions` because it's not obvious that we should. These functions just happen to have the same functionality right now. However, in the future, we might have macro properties and might want to use separate schemas for validating macro properties and macro definitions. In either case, removing `get_schema_for_macro_edge_definitions` should be done in it's own PR.

Reply:
Why would we remove `get_schema_for_macro_edge_definitions`? Based on your point that `[...] in the future, we might have macro properties and might want to use separate schemas for validating macro properties and macro definitions` shouldn't we remove `get_schema_for_macro_definition` instead and keep functions specific to each particular kind of macro?

Reply:
I am not sure that we should remove `get_schema_for_macro_definition`. You might want a schema with all macro definition directives included when building a UI for registering macro definitions. 

The point I am trying to make is that it's not obvious what to do here, so whatever we end up doing should be done in a separate PR.